pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -164,7 +164,7 @@ type PriorityQueue struct {
 
 	// inFlightPods holds the UID of all pods which have been popped out for which Done
 	// hasn't been called yet - in other words, all pods that are currently being
-	// processed.
+	// processed (being scheduled, in permit, or in the binding cycle).
```

Comment:
I won't block on this, but, this could be a blocker for ever having in-tree Permit plugins.

If a Permit plugin can take arbitrarily long, the list of inFlightPods and events might cause the scheduler to crash. Perhaps there should be a way for the framework to cancel a pod that is waiting on Permit.